packages/e2e/test/wallet/SingleAddressWallet/voting.test.ts

```diff
@@ -0,0 +1,93 @@
+  let wallet: SingleAddressWallet;
+  let keyAgent: KeyManagement.AsyncKeyAgent;
+
+  beforeAll(async () => {
+    keyAgent = await keyAgentById(0, env.KEY_MANAGEMENT_PROVIDER, env.KEY_MANAGEMENT_PARAMS);
+    wallet = new SingleAddressWallet(
+      { name: 'Test Wallet' },
+      {
+        assetProvider: await assetProviderFactory.create(env.ASSET_PROVIDER, env.ASSET_PROVIDER_PARAMS),
+        chainHistoryProvider: await chainHistoryProviderFactory.create(
+          env.CHAIN_HISTORY_PROVIDER,
+          env.CHAIN_HISTORY_PROVIDER_PARAMS
+        ),
+        keyAgent,
+        networkInfoProvider: await networkInfoProviderFactory.create(
+          env.NETWORK_INFO_PROVIDER,
+          env.NETWORK_INFO_PROVIDER_PARAMS
+        ),
+        rewardsProvider: await rewardsProviderFactory.create(env.REWARDS_PROVIDER, env.REWARDS_PROVIDER_PARAMS),
+        stakePoolProvider: await stakePoolProviderFactory.create(
+          env.STAKE_POOL_PROVIDER,
+          env.STAKE_POOL_PROVIDER_PARAMS
+        ),
+        txSubmitProvider: await txSubmitProviderFactory.create(env.TX_SUBMIT_PROVIDER, env.TX_SUBMIT_PROVIDER_PARAMS),
+        utxoProvider: await utxoProviderFactory.create(env.UTXO_PROVIDER, env.UTXO_PROVIDER_PARAMS)
+      }
+    );
+  });
+
+  afterAll(() => wallet.shutdown());
+
+  test('can submit tx with voting metadata and then query it', async () => {
+    const fullDerivationPath = [1852, 1815, 0, 0, 0];
+    const governanceApi = new GovernanceApiBase(keyAgent, keyAgent.exportKeyPair.bind(keyAgent));
+    const rewardAccount = (await firstValueFrom(wallet.addresses$))[0].rewardAccount;
+    const ownAddress = (await firstValueFrom(wallet.addresses$))[0].address;
+    const votingKeyPair = await governanceApi.getVotingKey(fullDerivationPath);
+    const auxiliaryData = await governanceApi.buildDelegation({
+      nonce: 1_234_567,
+      rewardAccount,
+      votingPublicKey: votingKeyPair.vkey
+    });
+    const txInternals = await wallet.initializeTx({
+      auxiliaryData,
+      outputs: new Set([{ address: ownAddress, value: { coins: 1_000_000n } }])
+    });
+    const outgoingTx = await wallet.finalizeTx(txInternals, auxiliaryData);
+    await wallet.submitTx(outgoingTx);
+    const loadedTx = await firstValueFrom(
+      wallet.transactions.history$.pipe(
+        map((txs) => txs.find((tx) => tx.id === outgoingTx.id)),
+        filter(isNotNil)
+      )
+    );
+    const auxBlobData = auxiliaryData.body.blob?.get(BigInt(VotingLabels.DATA)) as Cardano.MetadatumMap;
+    const loadedTxAuxBlobData = loadedTx.auxiliaryData?.body.blob?.get(
+      BigInt(VotingLabels.DATA)
+    ) as Cardano.MetadatumMap;
+    const auxBlobSignature = auxiliaryData.body.blob?.get(BigInt(VotingLabels.SIG)) as Cardano.MetadatumMap;
+    const loadedTxAuxBlobSignature = loadedTx.auxiliaryData?.body.blob?.get(
+      BigInt(VotingLabels.SIG)
+    ) as Cardano.MetadatumMap;
+    const votingSignature = auxBlobData.get(1n) as Buffer;
```

Comment:
`Buffer` is `Uint8Array`, but better use the narrower type here that is used in `core` [type](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Cardano/types/AuxiliaryData.ts#L9).
```suggestion
    const votingSignature = auxBlobData.get(1n) as Uint8Array;
```